code/web/src/modules/profileForm/ProfileForm.js

```diff
@@ -0,0 +1,97 @@
+import React from 'react';
+import { Component } from 'react';
+import { connect } from 'react-redux';
+import Button from '../../ui/button';
+import { saveProfile } from '../user/api/actions';
+import { changeEditMode } from '../user/api/actions';
+import Availability from '../availability/Availability';
+
+
+// import { updateAvailability } from '../subscription/api/actions';
+
+class ProfileForm extends Component{
+  constructor(props) {
+    super(props)
+
+      this.state = {
+        img: '',
+        address: '',
+        email: '',
+        description: ''
+      }
+  }
+
+  updateProfileState(e) {
+    this.setState({ [e.target.name]: e.target.value })
+  }
+
+  updateProfile = () => {
+    let newState = {
+          img: this.state.img || this.props.user.details.img || '',
+          address: this.state.address || this.props.user.details.address || '',
+          email: this.state.email || this.props.user.details.email || '',
+          description: this.state.description || this.props.user.details.description || ''
+        }
+    this.props.saveProfile(newState)
+    this.props.changeEditMode(this.props.user)
+  }
```

Comment:
The `newState` object allows us to set the value of each key in state based on true or false conditions.  Lines 30 - 33 say "If the user has typed in a new value for the input (address, email, etc), assign that value.  If not, check for a value already existing in props (eventually this will come from an `axios.post` request).   If the user has never submitted a value, set it to an empty string, as is the default.  We will likely add some error handling here to edit this code; removing `|| ' '` and instead, prompting the user to submit a value for each input.  

We then pass this `newState` object to `saveProfile`, which is an action creator from `./actions.js` that then communicates with all the data to update the user object with the new information.  

Hope that's helpful!

Reply:
Just a note for the future, with the addition of the description, image, and address fields to our backend, the final `|| ''` may become redundant. `this.props.user.details.x` will be null and I don't see a need to really replace it with an empty string unless I am missing something. Just a thought